### PR TITLE
Add train_subprocess for more stable training

### PR DESCRIPTION
**Pull request by MatthewScholefield**
This allows training in a subprocess to ensure all threads get shut down properly.

### CODE REVIEW

File: padatious/__main__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    else:
+        for fn in args.input_files:
+            obj_name, ext = splitext(basename(fn))
+            if ext == '.intent':
+                cont.load_intent(obj_name, fn)
+            elif ext == '.entity':
+                cont.load_entity(obj_name, fn)
+            else:
+                parser.error('Unknown file extension: {}'.format(ext))
+    kwargs = inspect.signature(cont.train).bind(*args.args).arguments
```
Comment (forslund) [inline padatious/__main__.py]: This seem to require that arguments are provided or atleast `-a []` is provided. Not sure it makes much of difference in the intended way to run this?

**Review by forslund — COMMENTED** [review]
I tried to get it to generate some output but failed, and I might have misunderstood the arguments required. The program would error if I didn't include the -a parameter

My command line was:
`python -m padatious train -a [] ./pad/ /opt/mycroft/skills/spotify-skill/locale/en-us/ShuffleOff.intent`

No cache files appeared in the pad subdirectory.

Running mycroft normally seems to work like previously.

File: padatious/__main__.py (hunk 1 above)
Comment (MatthewScholefield) [reply padatious/__main__.py]: Fixed. Yeah, this was an error with the default value of an argument.

**Comment by MatthewScholefield** [comment]
Ok, should be fixed now. Turns out there was a bug with `load_intent` and `load_entity` not setting the `must_train` flag.

**Comment by forslund** [comment]
Nice got output now. Merging

**Pull request merged**